### ISSUE TITLE
GUI: enhancement of Stacked Tokens

### ISSUE THREAD

**Issue by thefnitalian** [magefree/mage#7262]
Implementation of one token representing multiple of the same tokens in a stacked view similar to arena with a number on the token representing how many of that similar token there are similar to the attached link. https://i.redd.it/41t3vyvof5z01.png

**Comment by Woogachaka** [comment]
This would be a tremendous help/improvement. As a krenko-EDH player, I would love to see this feature.

**Comment by thefnitalian** [comment]
I haven’t seen any work done on this enhancement and it’s almost over two years old.

**Comment by jeffwadsworth** [comment]
Because a dev with knowledge of this particular code would have to dive in to fix it.  No one has, but you never know.

**Comment by thefnitalian** [comment]
I am going to close this as there seems no interest in this enhancement ever being followed up on.

**Comment by thefnitalian** [comment]
Some time has passed so I’m reopening this issue with perhaps just adding in option under settings that is a checkbox for stacking treasures, clues, etc?

**Comment by xenohedron** [comment]
The relevant code appears to be in `CardPluginImpl::sortPermanents`

**Comment by JayDi85** [comment]
Stacking cards feature is not about hard code — its small part and already implemented by the lands (I refactor/fix it while works on card icons/panels — you can enable it for other cards by “few lines”).

The main problem in UX — you must introduce new “gui container” logic to work with it (fake card object with useful icons and controls like popup window for more precise selections). It also can help to increase stack limit from x5 to infinite amount per stack (current x5 limit introduced due small free space on battlefield for a cards row);

Lands stacking can ignore that UX problem due lands role — rare usage in user interactions, only taps (e.g. you need only small space for mouse click). Other cards require much more interactions and fast info like P/T — stacking can hide it and make life harder for a player. Even MTGA does not solve it fully — those  card moving and auto-sorting on targeting are looks bad.

So the main idea behind stacked feature — prepare special GUI for it. BTW it will allow to stack user selected objects too.

**Comment by xenohedron** [comment]
I agree that for the general case of stacking any sort of permanent, a new gui container of some sort would be needed for UX.

However, there are some specific predefined tokens (rule [111.10](https://yawgatog.com/resources/magic-rules/#R11110)) that have similar role to lands, without UX problem when stacking. Perhaps for now we could implement stacking specifically for Treasure, Clue, Food, Blood, Powerstone tokens?

**Comment by JayDi85** [comment]
Yes, it can applied to such tokens.

**Comment by thefnitalian** [comment]
Does that mean that we may see a config option in the future to allow for stacking of predifined tokens?

**Comment by JayDi85** [comment]
Possible alternative implementation without GUI rework, only HumanPlayer improve:

![IMG_0190](https://github.com/magefree/mage/assets/8344157/d0d98245-57a4-48cd-9bfa-e5397b5d79a8)

**Comment by thefnitalian** [comment]
Did this ever get implemented? This is 4 years old

**Comment by JayDi85** [comment]
@thefnitalian yep, soon
